pkg/util/equality/podset.go

```diff
@@ -26,30 +26,33 @@ import (
 
 // TODO: Revisit this, maybe we should extend the check to everything that could potentially impact
 // the workload scheduling (priority, nodeSelectors(when suspended), tolerations and maybe more)
-func comparePodTemplate(a, b *corev1.PodSpec) bool {
+func comparePodTemplate(a, b *corev1.PodSpec, checkCount, changePodSpecFields bool) bool {
+	if changePodSpecFields && !equality.Semantic.DeepEqual(a.Tolerations, b.Tolerations) {
```

Comment:
why do we need this boolean?

Reply:
the tolerations could change when the job is unsuspended